text/php/0002-buildpack-yml-to-env-var.md

```diff
@@ -0,0 +1,222 @@
+```shell
+$BP_REDIS_SESSION_STORE_SERVICE_NAME="redis-session"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ redis:
+   session_store_service_name: redis-sessions
+```
+
+### `BP_MEMCACHED_SESSION_STORE_SERVICE_NAME`
+```shell
+$BP_MEMCACHED_SESSION_STORE_SERVICE_NAME="memcached-session"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ memcached:
+   session_store_service_name: memcached-sessions
+```
+
+### Configuration Removal
+The following structure in `buildpack.yml` will not be receiving an environment
+variable configuration option:
+```yaml
+php:
+ script:
+```
+This structure was effectively taking the part and behavior of [Paketo Procfile
+buildpack](https://github.com/paketo-buildpacks/procfile) by allowing users to
+provide a start command in `buildpack.yml`. However, as part of the [PHP Rewrite RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/php/0001-restructure.md)
+the Procfile buildpack will be added to all order groupings. Because of this
+addition I am proposing the removal of this configuration option in the
+environment variables in favor of encouraging users to use a Procfile if they
+need to set a custom start script.
+
+---
+The proposed environment variables for Composer are as follows:
+
+### `BP_COMPOSER_VERSION`
+```shell
+$BP_COMPOSER_VERSION="1.10.x"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  version: 1.10.x
+```
+### `BP_COMPOSER_INSTALL_OPTIONS`
+```shell
+$BP_COMPOSER_INSTALL_OPTIONS="--no-dev --prefer-install=auto"
+```
+
+Note: This will be parsed using this [shellwords library](https://github.com/mattn/go-shellwords).
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  install_options: ["--no-dev", "--prefer-install=auto"]
+```
+### `BP_COMPOSER_VENDOR_DIR`
+```shell
+$BP_COMPOSER_VENDOR_DIR="vendor"
+```
+
+Note: The value of this environment variable will be used to set
+[`$COMPOSER_VENDOR_DIR`](https://getcomposer.org/doc/03-cli.md#composer-vendor-dir)
+therefore if a user attempts to set this value themselves as well as set the
+value of `$BP_COMPOSER_VENDOR_DIR` the value of `$COMPOSER_VENDOR_DIR` will be
+overwritten resulting in potentially unexpected behavior.
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  vendor_directory: vendor
+```
+
+### `BP_COMPOSER_HOME`
+```shell
+$BP_COMPOSER_HOME="composer"
+```
+
+Note: The value of this environment variable will be used to set
+[`$COMPOSER_HOME`](https://getcomposer.org/doc/03-cli.md#composer-home)
+therefore if a user attempts to set this value themselves as well as set the
+value of `$BP_COMPOSER_HOME` the value of `$COMPOSER_HOME` will be overwritten
+resulting in potentially unexpected behavior.
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  json_path: composer
+```
```

Comment:
Same thoughts here. If all we're doing is setting `COMPOSER_HOME` then we should probably drop this config value and instruct users to set `COMPOSER_HOME` directly for this functionality.

Reply:
I think that this has the same potential drawback as the environment variable above so I think whatever decision we make there we should copy for consistency sake.

Reply:
Actually, for this one I'm not sure that `$COMPOSER_HOME` is the right match. `$COMPOSER_HOME` is changing the global configuration directory. This is where composer loads defaults from. While json_path was meant to allow the user to change where the composer.json file is located.

I think it more closely correlates to the `$COMPOSER` setting, which tells COMPOSER where to find composer.json. We were not using that setting previously though. Previously, we'd "find" the composer.json using the path trick above and then launch `composer` processes from within that directory (i.e. set the working directory). 

It's probably better to drop that and support using `$COMPOSER` for that. 

Setting `$COMPOSER_HOME` is something the buildpack will need to do and something we'll discourage users from doing. I suspect that stuff will go into a layer so it's cached from run to run. The current buildpack is doing that.

Reply:
Thanks for this info. I completely missed the `$COMPOSER` environment variable in my reading. I will make the switch and update it accordingly because I am a big fan of using native configuration.

Reply:
Alright this has been reworked and should be more in line with what you were talking about.